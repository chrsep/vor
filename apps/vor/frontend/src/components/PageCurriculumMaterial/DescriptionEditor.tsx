import { Trans } from "@lingui/macro"
import React, { FC, useState } from "react"
import { Button, Card, Flex } from "theme-ui"
import usePatchMaterial from "../../hooks/api/curriculum/usePatchMaterial"
import { ReactComponent as CloseIcon } from "../../icons/close.svg"
import Icon from "../Icon/Icon"
import MarkdownEditor from "../MarkdownEditor/MarkdownEditor"
import { Typography } from "../Typography/Typography"

const DescriptionEditor: FC<{
  initialValue?: string
  onDismiss: () => void
  onSave: () => void
  materialId: string
  subjectId: string
}> = ({ materialId, subjectId, onDismiss, onSave, initialValue = "" }) => {
  const [value, setValue] = useState(initialValue)
  const patchMaterial = usePatchMaterial(materialId, subjectId)

  const handleSave = async () => {
    try {
      const result = await patchMaterial.mutateAsync({ description: value })
      if (result?.ok) onSave()
      onSave()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <Card variant="responsive">
      <Flex pt={3} pb={2}>
        <Typography.Body px={3} sx={{ fontWeight: "bold" }}>
          <Trans>Description</Trans>
        </Typography.Body>

        <Button variant="outline" ml="auto" p={1} mr={2} onClick={onDismiss}>
          <Icon as={CloseIcon} fill="danger" />
        </Button>

        <Button
          mr={3}
          sx={{ fontWeight: "bold", fontSize: 0 }}
          onClick={handleSave}
        >
          <Trans>Save</Trans>
        </Button>
      </Flex>

      <MarkdownEditor value={value} onChange={setValue} />
    </Card>
  )
}

export default DescriptionEditor
