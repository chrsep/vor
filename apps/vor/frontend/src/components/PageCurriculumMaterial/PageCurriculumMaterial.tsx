import React, { FC } from "react"
import { Box, Button, Card, ThemeUIStyleObject } from "theme-ui"
import { useGetArea } from "../../hooks/api/useGetArea"
import { useGetSubject } from "../../hooks/api/useGetSubject"
import { ReactComponent as DeleteIcon } from "../../icons/trash.svg"
import {
  ADMIN_CURRICULUM_URL,
  ADMIN_URL,
  CURRICULUM_AREA_URL,
  CURRICULUM_SUBJECT_URL,
} from "../../routes"
import DataBox from "../DataBox/DataBox"
import Icon from "../Icon/Icon"
import MultilineDataBox from "../MultilineDataBox/MultilineDataBox"
import TopBar, { breadCrumb } from "../TopBar/TopBar"

export interface PageCurriculumMaterialProps {
  sx?: ThemeUIStyleObject
  areaId: string
  subjectId: string
  materialId: string
}
const PageCurriculumMaterial: FC<PageCurriculumMaterialProps> = ({
  subjectId,
  areaId,
}) => {
  const area = useGetArea(areaId)
  const subject = useGetSubject(subjectId)

  return (
    <Box mx="auto" sx={{ maxWidth: "maxWidth.lg", width: "100%" }}>
      <TopBar
        containerSx={{ display: ["flex", "flex", "none"] }}
        breadcrumbs={[
          breadCrumb("Admin", ADMIN_URL),
          breadCrumb("Curriculum", ADMIN_CURRICULUM_URL),
          breadCrumb(area.data?.name ?? "", CURRICULUM_AREA_URL(areaId)),
          breadCrumb(
            subject.data?.name ?? "",
            CURRICULUM_SUBJECT_URL(areaId, subjectId)
          ),
          breadCrumb("Material"),
        ]}
      />

      <Box p={3}>
        <Card sx={{ width: "100%" }}>
          <DataBox label="Material Name" value="..." />
        </Card>

        <Card mt={3} pb={2}>
          <MultilineDataBox label="Description" value="" placeholder="..." />
        </Card>

        <Button variant="outline" sx={{ color: "danger" }} ml="auto" mt={3}>
          <Icon as={DeleteIcon} mr={2} fill="danger" />
          Delete
        </Button>
      </Box>
    </Box>
  )
}

export default PageCurriculumMaterial
