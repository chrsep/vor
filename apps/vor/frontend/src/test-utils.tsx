/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { setupI18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import Theme from "./gatsby-plugin-theme-ui"
import enCatalog from "../i18n/lingui/en/messages.js"

const i18n = setupI18n()
// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
