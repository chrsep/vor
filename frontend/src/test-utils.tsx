/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <Box fontSize={[16, 20]}>{children}</Box>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
