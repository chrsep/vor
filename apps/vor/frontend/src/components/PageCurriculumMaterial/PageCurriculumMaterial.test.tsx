import React from "react"
import { render } from "../../test-utils"
import PageCurriculumMaterial from "./PageCurriculumMaterial"

describe("PageCurriculumMaterial", () => {
  it("should render correctly", () => {
    const { container } = render(<PageCurriculumMaterial />)
    expect(container).toMatchSnapshot()
  })
})
