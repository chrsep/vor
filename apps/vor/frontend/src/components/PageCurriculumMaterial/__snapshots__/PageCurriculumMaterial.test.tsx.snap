// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCurriculumMaterial should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 900px;
  width: 100%;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width: 40em) {
  .emotion-2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width: 52em) {
  .emotion-2 {
    display: none;
  }
}

.emotion-3 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary, #00e399);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary, #027c5b);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter, rgba(0,227,153,0.12));
  border-color: var(--theme-ui-colors-primary, #00e399);
  outline: none;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon, rgba(0,0,0,0.6));
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-7 {
  display: inline-block;
  font-size: 12px;
  color: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
  white-space: nowrap;
}

.emotion-7:last-child {
  color: var(--theme-ui-colors-textPrimary, #027c5b);
}

.emotion-7:last-child>.splitter {
  display: none;
}

.emotion-8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 12px;
  color: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
  white-space: nowrap;
}

.emotion-19:last-child {
  color: var(--theme-ui-colors-textPrimary, #027c5b);
}

.emotion-19:last-child>.splitter {
  display: none;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface, #fff);
  border-radius: 6px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  width: 100%;
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text, #37383c);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 4px;
  color: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
  line-height: 1;
  font-size: 12px;
}

.emotion-25 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text, #37383c);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
}

.emotion-26 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary, #00e399);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary, #027c5b);
  background-color: var(--theme-ui-colors-surface, #fff);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: auto;
  padding-left: 8px;
  padding-right: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-26:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-26:hover,
.emotion-26:focus {
  background-color: var(--theme-ui-colors-primaryLighter, rgba(0,227,153,0.12));
  border-color: var(--theme-ui-colors-primary, #00e399);
  outline: none;
}

.emotion-26:hover:disabled,
.emotion-26:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
}

.emotion-27 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon, rgba(0,0,0,0.6));
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-28 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface, #fff);
  border-radius: 6px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  margin-top: 16px;
  padding-bottom: 8px;
}

.emotion-29 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-30 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-31 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text, #37383c);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin: 16px;
  font-weight: bold;
}

.emotion-32 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary, #00e399);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary, #027c5b);
  background-color: var(--theme-ui-colors-surface, #fff);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: auto;
  margin-right: 16px;
  padding: 8px;
  color: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
}

.emotion-32:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-32:hover,
.emotion-32:focus {
  background-color: var(--theme-ui-colors-primaryLighter, rgba(0,227,153,0.12));
  border-color: var(--theme-ui-colors-primary, #00e399);
  outline: none;
}

.emotion-32:hover:disabled,
.emotion-32:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
}

.emotion-34 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
}

.emotion-35 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary, #00e399);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary, #027c5b);
  background-color: var(--theme-ui-colors-surface, #fff);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: auto;
  margin-top: 16px;
  color: var(--theme-ui-colors-danger, #d50000);
}

.emotion-35:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-35:hover,
.emotion-35:focus {
  background-color: var(--theme-ui-colors-primaryLighter, rgba(0,227,153,0.12));
  border-color: var(--theme-ui-colors-primary, #00e399);
  outline: none;
}

.emotion-35:hover:disabled,
.emotion-35:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
}

.emotion-36 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: 8px;
  fill: var(--theme-ui-colors-danger, #d50000);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          data-cy="back-button"
          href="/dashboard/admin/curriculum/subjects?areaId=&subjectId="
        >
          <button
            class="emotion-4"
          >
            <div
              class="emotion-5"
            />
          </button>
        </a>
        <div
          class="emotion-6"
        >
          <a
            class="emotion-7"
            href="/dashboard/admin"
          >
            <span
              class="emotion-8"
            >
              Admin
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/admin/curriculum"
          >
            <span
              class="emotion-8"
            >
              Curriculum
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/admin/curriculum/area?areaId="
          >
            <span
              class="emotion-8"
            />
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/admin/curriculum/subjects?areaId=&subjectId="
          >
            <span
              class="emotion-8"
            />
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <div
            class="emotion-19"
          >
            Material
            <span
              class="splitter"
            >
              /
            </span>
          </div>
        </div>
      </div>
      <div
        class="emotion-20"
      >
        <div
          class="emotion-21"
        >
          <div
            class="emotion-22"
          >
            <div
              class="emotion-23"
            >
              <p
                class="emotion-24"
              >
                Material Name
              </p>
              <p
                class="emotion-25"
              >
                ...
              </p>
            </div>
            <button
              aria-label="edit-material-name"
              class="emotion-26"
            >
              <div
                class="emotion-27"
              />
            </button>
          </div>
        </div>
        <div
          class="emotion-28"
        >
          <div
            class="emotion-29"
          >
            <div
              class="emotion-30"
            >
              <p
                class="emotion-31"
              >
                Description
              </p>
              <button
                aria-label="edit-description"
                class="emotion-32"
                data-cy="edit-description"
              >
                <div
                  class="emotion-27"
                />
              </button>
            </div>
            <div
              class="prose prose-sm max-w-none undefined emotion-34"
            >
              <p>
                ...
              </p>
            </div>
          </div>
        </div>
        <button
          class="emotion-35"
        >
          <div
            class="emotion-36"
          />
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
